### PR TITLE
code of teamlist

### PR DESCRIPTION
**Pull request by vivian0420**
## Pull Request
Added /teams router and TeamList Component

### Description
Commit: code of teamlist:
1. Built /teams router under /[competition]
2. Created TeamList Component
3. Implemented the URL query parameter 'season={id}' under the teams router

Commit: refactored the TeamCard component from the TeamList component:
1. Created TeamCard component from the TeamList component

Commit: Fixed the team logo uploading issues:
1. If a team logo uploading issue occurs, utilize the first letter of the team name to replace the team logo.

Commit: Added key prop for each teamCard:
1. Added key prop to teamCard to get rid of the error => 'missing "key" prop for element in iterator'

### CODE REVIEW

File: app/basketball/[competition]/teams/page.tsx [hunk 1]
```diff
@@ -0,0 +1,45 @@
+import Custom404 from "@/components/404";
+import SeasonDropDownMenu from "@/components/basketball/teams/TeamDropDowmMenu";
+import TeamList from "@/components/basketball/teams/TeamList";
+import { asyncFetch } from "@/utils/fetch";
+import { getRecentSeasonByGroup } from "@/utils/get-recent-seasons";
+import { GROUPNAME_TO_COMPETITIONID } from "@/utils/variables";
+
+export default async function Page({ params, searchParams }: any) {
+  // fetch current season
+  const season = await getRecentSeasonByGroup(params.competition);
```
Comment (umliuxin) [inline app/basketball/[competition]/teams/page.tsx]: the ordering of this request is introducing useless requests
supppose to be sth like
```
if(season exists in queryParam){
fetch season by id
} else {
fetch reason season
}

File: components/basketball/teams/TeamCard.tsx [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+  return (
+    <Card key={team.id} className="relative" isPressable>
+      <CardBody className="justify-center items-center h-40 hover:opacity-50">
+        <Image
+          width={100}
+          height={100}
+          alt={team.name}
+          src={team.logosrc}
+          onError={() => {
+            setVisibility("visible");
```
Comment (umliuxin) [inline components/basketball/teams/TeamCard.tsx]: nit, update the state to be more descriptive about image loading

File: components/basketball/teams/TeamCard.tsx [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+  team: any;
+}
+
+const TeamCard: React.FC<TeamCardProp> = ({ team }) => {
+  const [visibility, setVisibility] = useState("invisible");
+  return (
+    <Card key={team.id} className="relative" isPressable>
+      <CardBody className="justify-center items-center h-40 hover:opacity-50">
+        <Image
+          width={100}
```
Comment (umliuxin) [inline components/basketball/teams/TeamCard.tsx:15]: let's use lazy here for better performance

File: components/basketball/teams/TeamCard.tsx [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+          height={100}
+          alt={team.name}
+          src={team.logosrc}
+          onError={() => {
+            setVisibility("visible");
+          }}
+        />
+        <p className={`${visibility} absolute font-medium text-6xl`}>{team.shortname.substring(0,1).toUpperCase()}</p>
```
Comment (umliuxin) [inline components/basketball/teams/TeamCard.tsx]: prefer to be conditional

**Comment by umliuxin** [comment]
Build is failing due to some issues like 
```
i [Error]: Page with `dynamic = "error"` couldn't be rendered statically because it used `searchParams.season`.
    at s (/github/workspace/.next/server/chunks/248.js:1:98315)
    at Object.get (/github/workspace/.next/server/chunks/248.js:1:97442)
    at m (/github/workspace/.next/server/app/basketball/[competition]/teams/page.js:1:[566](https://github.com/svcsa-dev/svcsa-react/actions/runs/8806973560/job/24172961171#step:4:567)2) {
  code: 'NEXT_STATIC_GEN_BAILOUT'
}
```

This is due to some issue in the SSR mode. In the build time, the app will generate static pages, which is not working with search params.

I am not expert in this either, encourage you to try to figure out the Next.js doc to see if you can solve it

**Comment by vivian0420** [comment]
I have addressed the build errors and optimized the code according to the comments provided earlier. Please review the changes.

**Review by umliuxin — APPROVED** [review]

**Pull request merged**